src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -486,7 +488,40 @@ ElementItem* ElementItemDelegate::getElementtItem(const QModelIndex& index) cons
     ElementView* elementView = static_cast<ElementView*>(parent());
     return elementView->itemFromIndex(index);
 }
-// ----------------------------------------------------------------------------
+
+/* Filter element list widget ------------------------------------------------------ */
+FilterList::FilterList(QWidget* parent) : QListWidget(parent)
+{
+    {
+        QSignalBlocker block(this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", "Normal"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", "Construction"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", "Internal"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", "External"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", "All types"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Point"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Line"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Circle"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Ellipse"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Arc"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Arc of ellipse"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Arc of hyperbola"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - Arc of parabola"), this);
+        new QListWidgetItem(QApplication::translate("ConstraintFilterList", " - B-Spline"), this);
+
+        for (int i = 0; i < count(); i++) {
+            QListWidgetItem* it = item(i);
+
+            it->setFlags(it->flags() | Qt::ItemIsUserCheckable);
+            it->setCheckState(Qt::Checked);
+        }

```

Comment:
1. Ok the context needs to be class name. I didn't knew. I changed it to ElementFilterList then. As 'constraintFilterList' was a copy paste from constraint PR.
2. So I change to : 
```
addItem(new QListWidgetItem(QApplication::translate("ElementFilterList", "Normal"), this));
    addItem(new QListWidgetItem(QApplication::translate("ElementFilterList", "Construction"), this));
    addItem(new QListWidgetItem(QApplication::translate("ElementFilterList", "Internal"), this));
```
Is that right?
3. regarding the language. You mean that if my FC is in english, I open a sketch, go to preference change language. Then the text don't translate unless you close sketch and reopen? That doesn't sound like such a big issue. Relaunching an app after changing language is acceptable. Not sure how to do without that?

Reply:
1.
2.
No that's not it. But I will provide snippet when 3 is discussed. ;) 

3. I'm not sure it's so easily acceptable because there is currently an issue open for this exact topic. :)

Reply:
I understand. But then the problem maybe out of scope of the PR? Or do we need to fix the syntax here somehow?